src/ApplicationCore/Services/OrderService.cs

```diff
@@ -30,9 +30,9 @@ public class OrderService : IOrderService
     public async Task CreateOrderAsync(int basketId, Address shippingAddress)
     {
         var basketSpec = new BasketWithItemsSpecification(basketId);
-        var basket = await _basketRepository.GetBySpecAsync(basketSpec);
+        var basket = await _basketRepository.FirstOrDefaultAsync(basketSpec);
 
-        Guard.Against.NullBasket(basketId, basket);
+        Guard.Against.Null(basket, nameof(basket));
```

Comment:
If you're refactoring away from NullBasket make sure you delete it from Extensions/GuardExtensions as well.